src/TrackPanel.cpp

```diff
@@ -735,11 +742,62 @@ void TrackPanel::OnMouseEvent(wxMouseEvent & event)
          }
       } );
    }
+    
+    if(event.RightDown())
+    {
+        wxWindow *parent = wxWindow::FindFocus();
+        if(!parent)
+           parent = &GetProjectFrame( *GetProject() );
+        
+        if(parent)
+        {
+            wxMenu contextMenu;
+            contextMenu.Bind(wxEVT_MENU,
+               [this]( wxCommandEvent &event ){
+                  OnContextMenu( /*GetProject(),*/ event ); }
+            );
+            
+            contextMenu.Append(kContextMenuItemID_AddMonoTrack, XO("Add mono track").Translation());
+            contextMenu.Append(kContextMenuItemID_AddStereoTrack, XO("Add stereo track").Translation());
+            contextMenu.Append(kContextMenuItemID_AddLabelTrack, XO("Add label track").Translation());
+            contextMenu.AppendSeparator();
+            contextMenu.Append(kContextMenuItemID_ExportProject, XO("Export project\tCtrl+E").Translation());
+            contextMenu.AppendSeparator();
+            contextMenu.Append(kContextMenuItemID_SelectAll, XO("Select All\tCtrl+A").Translation());
+            
+            parent->PopupMenu(&contextMenu);
+        }
+    }
 
    // Must also fall through to base class handler
    event.Skip();
 }
 
+#include "SelectUtilities.h"
+
+void TrackPanel::OnContextMenu(/*AudacityProject &project, */ wxCommandEvent& event)
+{
```

Comment:
I don't like making TrackPanel the class that handles this.  Maybe it should be done within class BackgroundCell instead, by analogy with how LabelTrackView handles context menus.